src/rshapes.c

```diff
@@ -197,19 +198,36 @@ void DrawLineBezier(Vector2 startPos, Vector2 endPos, float thick, Color color)
     Vector2 previous = startPos;
     Vector2 current = { 0 };
 
+    Vector2* pts = RL_MALLOC((2*BEZIER_LINE_DIVISIONS+2)*sizeof(Vector2));
+
     for (int i = 1; i <= BEZIER_LINE_DIVISIONS; i++)
     {
         // Cubic easing in-out
         // NOTE: Easing is calculated only for y position value
         current.y = EaseCubicInOut((float)i, startPos.y, endPos.y - startPos.y, (float)BEZIER_LINE_DIVISIONS);
         current.x = previous.x + (endPos.x - startPos.x)/ (float)BEZIER_LINE_DIVISIONS;
 
-        // TODO: Avoid drawing the line by pieces, it generates gaps for big thicks,
-        // Custom "triangle-strip" implementation should be used, check DrawTriangleStrip() for reference
-        DrawLineEx(previous, current, thick, color);
+        double dy = current.y-previous.y;
+        double dx = current.x-previous.x;
+        double size = 0.5*thick/sqrt(dx*dx+dy*dy);
```

Comment:
Should `dy`, `dx` and `size` be floats instead of doubles? 

`EaseCubicInOut` returns a float, and the values contained in `Vector2` are also floats.

Reply:
You are right, thanks for the review!

Reply:
> Should `dy`, `dx` and `size` be floats instead of doubles?
> 
> `EaseCubicInOut` returns a float, and the values contained in `Vector2` are also floats.

I suspect it is because the prototype of `sqrt` is

```C
double sqrt(double x)
```
and you don't want to lose precision going into or out of that function, especially because `dy` and `dx` are differences and those can be fragile if the differences between current and previous are relatively small.

Rather than simply line up types, it might be useful to do some test computations to see if there are noticeable differences in the actual precision achieved.

Reply:
I don't think it matters, 7 digits of precision are enough. I guess `size` can remain double but it's really nitpicky in my opinion.

Reply:
please, could you avoid doubles? I tried to avoid them in all raylib.

Reply:
It is valuable to keep in mind that differences of same-signed floating-point values are notorious for loss of _accuracy_, in some cases, regardless of the capacity (what we tend to call precision) of the floating-point values.  Then multiplying those differences by something (or squaring them) makes it worse.  (There's good reason that `sqrt` produces a `double` result.)

So, rather than believing in _x_-digits precision, one needs to be able to show how at least enough _accuracy_ is achieved.  In this case, enough to ensure smoothness in graphical determination.  Or simply hedge and use higher-precision for intermediate operations and their stored results.  The compiler and the floating-point instructions are already doing some of that without requiring any attention, so long as the code does not get in the way.

Reply:
> please, could you avoid doubles? I tried to avoid them in all raylib.

@raysan5 Do you use a compiler option to force that in intermediate calculations, or is your concern just about stored values and parameter types?  (I'm not certain the intermediate cases are avoidable.)